uml/backend-software-system/input-listener.puml

```diff
@@ -0,0 +1,53 @@
+@startuml input-listener-console
+package "Console" {
+
+    interface IEventListener {
+        +handler: IEventHandler
+        +native_listen(): List<String>
+    }
+
+    class KeyboardListener {
+        +handler: IEventHandler
+        +native_listen(): List<String>
+    }
+
+    class MouseListener {
+        +handler: IEventHandler
+        +native_listen(): List<String>
+    }
+
+    IEventListener <|.. KeyboardListener : implements
+    IEventListener <|.. MouseListener : implements
+
+    interface IEventHandler {
+        -poolEvents: Vec<String>
+        +handleEvent(): void
```

Comment:
What is the method for?

Reply:
This method decided how the events are handled by the handlers